source/run-command/run-command.rst

```diff
@@ -0,0 +1,185 @@
+``runCommand``
+--------------
+
+The following represents how a runCommand API SHOULD be exposed.
+
+.. code:: typescript
+
+    interface Db {
+      /**
+       * Takes an argument representing an arbitrary BSON document and executes it against the server.
+       */
+      runCommand(command: BSONDocument, options: RunCommandOptions): BSONDocument;
+    }
+
+    interface RunCommandOptions {
+      /**
+       * An optional readPreference setting to apply to server selection logic.
+       * This value MUST be applied to the command document as the $readPreference global command argument if not set to primary.
+       *
+       * @defaultValue primary
+       *
+       * @see https://www.mongodb.com/docs/manual/core/read-preference/
+       */
+      readPreference?: ReadPreference;
+
+      /**
+       * An optional explicit client session.
+       * The associated logical session id (`lsid`) MUST be applied to the command.
+       *
+       * @see https://www.mongodb.com/docs/manual/core/read-isolation-consistency-recency/#std-label-sessions
+       */
+      session?: ClientSession;
+
+      /**
+       * An optional timeout option to govern the amount of time that a single operation can execute before control is returned to the user.
+       * This timeout applies to all of the work done to execute the operation, including but not limited to server selection, connection checkout, and server-side execution.
+       *
+       * @ see https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/client-side-operations-timeout.rst
+       */
+      timeoutMS?: number;
+    }
+
+Deviations
+^^^^^^^^^^
+
+* Drivers MAY rename their API to match the language syntax.
+
+RunCommand implementation details
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+RunCommand provides a way to access MongoDB server commands directly without requiring a driver to implement a bespoke helper.
+The API is intended to take a document from a user and apply a number of common driver internal concerns before forwarding the command to a server.
+In general, a driver SHOULD avoid inspecting or modifying the user's command document in an effort to remain forward compatible with any potential server commands.
+
+Drivers that have historically modified user input SHOULD strive to instead clone the input such that appended fields do not affect the user's input.
+
+OP_MSG
+""""""
+
+The ``$db`` global command argument MUST be sent on the command sent to the server and it MUST correspond to the database name the runCommand API was invoked on.
+
+* See OP_MSG's section on `Global Command Arguments <https://github.com/mongodb/specifications/blob/master/source/message/OP_MSG.rst#global-command-arguments>`_
+
+ReadPreference
+""""""""""""""
+
+A driver's runCommand implementation MUST assume all commands are read operations.
+To facilitate selecting the server the command should be sent to a driver's runCommand MUST accept an optional Read Preference option.
```

Comment:
```suggestion
To facilitate server selection the runCommand operation MUST accept an optional Read Preference option.
```

I'm not sure if this suggestion conveys the original intention, but the existing text definitely needs some correction.